skybar/src/main/java/org/wtf/skybar/transform/SkybarMethodVisitor.java

```diff
@@ -1,29 +1,53 @@
 package org.wtf.skybar.transform;
 
+import org.objectweb.asm.Handle;
 import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
 import org.wtf.skybar.registry.SkybarRegistry;
 import org.wtf.skybar.transform.util.WorkingLineNumberVisitor;
 
+import java.lang.invoke.CallSite;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
 /**
  * Inserts instrumentation to update the SkybarRegistry on each line number.
  */
 class SkybarMethodVisitor extends WorkingLineNumberVisitor {
 
     private final String sourceFile;
+    private final int version;
 
-    public SkybarMethodVisitor(String sourceFile, MethodVisitor mv) {
+    public SkybarMethodVisitor(String sourceFile, int version, MethodVisitor mv) {
         super(ASM5, mv);
         this.sourceFile = sourceFile;
+        this.version = version;
     }
 
     @Override
     protected void onLineNumber(int lineNumber) {
-        final long index = SkybarRegistry.registry.registerLine(sourceFile, lineNumber);
+        SkybarRegistry.registry.registerLine(sourceFile, lineNumber);
+
+        if(useInvokeDynamic()) {
```

Comment:
This means we can only instrument code compiled with -target 1.7 or above, right? (Which I'm fine with, fwiw.)